src/Compilers/CSharp/Portable/Emitter/Model/PEModuleBuilder.cs

```diff
@@ -301,7 +301,10 @@ public sealed override IEnumerable<(Cci.ITypeDefinition, ImmutableArray<Cci.Debu
         /// </summary>
         private static void GetDocumentsForMethodsAndNestedTypes(PooledHashSet<Cci.DebugSourceDocument> documentList, ArrayBuilder<Cci.ITypeDefinition> typesToProcess, EmitContext context)
         {
-            Debug.Assert(!context.MetadataOnly);
+            // Temporarily disable assert to unblock getting net8.0 teststing re-nabled on Unix. Will 
+            // remove this shortly.
+            // https://github.com/dotnet/roslyn/issues/71571
+            // Debug.Assert(!context.MetadataOnly);
```

Comment:
@tmat FYI I need to temporarily disable this assert that you added in November 

https://github.com/dotnet/roslyn/commit/8920679b0517fec747e6d165980f611306cc6444#diff-de1a7398366a42226adb835505958a0e044f9b9816301dcfb013f09820d91cd9

This is blocking a number of compiler tests on Unix (only for some reason). This PR is needed before I can merge the move to `net8.0` which is why I'm going for temp disable vs. dig into this right now.

Reply:
Thanks. I keep meaning to file a bug on the fact that this is failing locally, only to get distracted by something else.

Reply:
Ok.